### PR TITLE
D3js (Not A Real Pull Request)

### PR DESCRIPTION
**Pull request by annathecrow**
This is not even remotely ready for... anything, but I thought you might want to have a look. I was trying if it will be possible to combine the jQuery code I wrote for the GUI with D3. Mostly because I was afraid that I would do a lot of GUI design and then had to rewrite half of it to accomodate the graphs. 
So I added D3 and cobbled together a _very simple_ graph to test things out... and it seems to be working. HA.
Es, do I understand correctly that you have experience with D3?
I would like to prepare the `home.js` in such a way that you could write the graphing code in a separate file and we'd keep the GUI and graphing apart. I would pretty much just hand over the data (maybe parsed into the format D3 requires for graphing) and that'd be it. IMO it'd be cleaner, and easier to maintain.  What do you think?

Also. I think this is our first official graph. :D

### CODE REVIEW

**Comment by esgibter** [comment]
ok i'm finally looking!

**Comment by esgibter** [comment]
> Es, do I understand correctly that you have experience with D3?

Yup.

> I would like to prepare the home.js in such a way that you could write the graphing code in a separate file and we'd keep the GUI and graphing apart. I would pretty much just hand over the data (maybe parsed into the format D3 requires for graphing) and that'd be it. IMO it'd be cleaner, and easier to maintain. What do you think?

Okay - that can work, it just gets weird when GUI interactions trigger API calls. I'm going to dig into your D3/jquery connections to explain more - I think right now what you're doing is calling the API via jquery, and then passing the data to D3. It might be cleaner to do the gui interactions with jquery, and the API calls with D3, since all of the API calls result in graph objects being rendered. 

We could also make this really simple and not use jquery for API-driven interactions at all - just gui interactions like show/hide stuff.

File: src/application/home/static/js/home.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    	
+    g.graph1.append("svg:text")
+    	.attr("x",margin.left)
+    	.attr("y",margin.top)
+    	.attr("text-anchor".top)
+    	.attr("transform","translate(0,18)")
+    	.text("Categories");
+};
+
+these = {
```
Comment (esgibter) [inline src/application/home/static/js/home.js]: what's this for? were you using it for testing?

Comment (annathecrow) [inline src/application/home/static/js/home.js]: that adds the labels (for the columns and the entire graph).
I'd say ignore the whole graph part, it's crap. We'll need to think about the graphs some more, anyway - do we want the graphs to fit the flexible width of the site or should they have a fixed size, horizontal or vertical, what kind of labels, that sort of thing.

**Comment by annathecrow** [comment]
> Okay - that can work, it just gets weird when GUI interactions trigger API calls.

Could you explain this more? I mean... that's what happens anyway, right? The user clicks, API call happens... I think I'm being dense, sorry.

> I think right now what you're doing is calling the API via jquery, and then passing the data to D3. 

Yeah, that's pretty much it. It made sense to me that way: when I handle the GUI, I need to do things depended on the status of the API call - change the button text to a loading icon, show/hide things on success, fill the textarea with raw output of the API. I thought letting jQuery be the "manager", prepare the canvases for the graphs and then leave on D3 only the graph drawing would make most sense. 
But if you think we should be doing it differently, we definitely can. There just needs to be some way to let the GUI know that the API call started/succeeded/failed so it can move things around properly.

**Pull request closed without merging**